test/AMAMM.t.sol

```diff
@@ -24,81 +25,136 @@ contract AmAmmTest is Test {
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, 1);
+        vm.stopPrank();
 
-    //     assertEq(amAmm.getManager(POOL_0, 0).bidder, address(user1));
+        assertEq(amAmm._getDeposit(POOL_0, 1), K * 1e18, "Bid Promoted to Top Bid");
 
-    //     vm.prank(user2);
-    //     vm.expectRevert();
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, K * 1e18, 0);
-    // }
+        vm.startPrank(user1);
+        amAmm.bidToken().approve(address(amAmm), 2e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 2e18, 10); // Future Bid
+        vm.stopPrank();
 
-    // //Test to check if lower bid can be promoted
-    // function test_withdrawFutureBid() external {
-    //     vm.prank(user0);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, K * 1e18, 0);
+        assertEq(amAmm.getManager(POOL_0, 10).bidder, address(user1)); //Confirm they are manager of future bid
 
-    //     assertEq(amAmm._getDeposit(POOL_0, 0), K * 1e18, "Bid Promoted to Top Bid");
+        vm.startPrank(user1);
+        vm.expectRevert();
+        amAmm.withdrawBalance(POOL_0, K * 10e18);
+        vm.stopPrank();
+    }
 
-    //     vm.prank(user1);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 10e18, K * 10e18, 10); // Future Bid
+    // Test if lower bids can be refunded
+    function test_bid_refund() external {
+        amAmm.bidToken().mint(address(user0), K * 100e18);
+        amAmm.bidToken().mint(address(user1), K * 100e18);
 
-    //     assertEq(amAmm.getManager(POOL_0, 10).bidder, address(user1)); //Confirm they are manager of future bid
+        vm.startPrank(user0);
+        amAmm.bidToken().approve(address(amAmm), 1e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, 1);
+        vm.stopPrank();
 
-    //     vm.prank(user1);
-    //     vm.expectRevert();
-    //     amAmm.withdrawBalance(POOL_0, K * 10e18);
-    // }
+        vm.startPrank(user1);
+        amAmm.bidToken().approve(address(amAmm), 10e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 10e18, 1); // New Bid Winner
+        vm.stopPrank();
 
-    // // Test if lower bids can be refunded
-    // function test_bid_refund() external {
-    //     vm.prank(user0);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, K * 1e18, 1);
+        assertEq(amAmm._getDeposit(POOL_0, 1), K * 10e18, "Bid Promoted to Top Bid");
 
-    //     vm.prank(user1);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 10e18, K * 10e18, 1); // New Bid Winner
+        skip(10800); //Enter Epoch 3
 
-    //     assertEq(amAmm._getDeposit(POOL_0, 1), K * 10e18, "Bid Promoted to Top Bid");
+        assertEq(amAmm._userBalance(user0), 1e18 * K, "Ensure User bid is still available to withdraw");
 
-    //     skip(10800); //Enter Epoch 3
+        vm.prank(user0); // user0 should have 1e18* k balance and should be able to withdraw
+        amAmm.withdrawBalance(POOL_0, 1e18 * K);
```

Comment:
It looks a bit confusing to me. Can we test with two different users so that user0 bid 1, then user1 bid 2, user2 becomes the winner and user0 would receive their refund so that `currentBalance` == `originalBalance`